### PR TITLE
Clarify datagram size authentication a bit

### LINKED ISSUE #4273: Re-writing for clarity
>  UDP datagrams MUST NOT be fragmented at the IP layer.  In IPv4
  ([IPv4]), the DF bit MUST be set if possible, to prevent
  fragmentation on the path.
>
>  Datagrams are required to be of a minimum size under some conditions.
  However, the size of the datagram is not authenticated.  Therefore,
  an endpoint MUST NOT close a connection when it receives a datagram
  that does not meet size constraints, though the endpoint MAY discard
  such datagrams.

I really think this section needs re-writing for clarity.

### PR DESCRIPTION
**Pull request by janaiyengar**
Fixes #4273.

### CODE REVIEW

File: draft-ietf-quic-transport.md [hunk 1]
```diff
@@ -4127,10 +4127,12 @@ UDP datagrams MUST NOT be fragmented at the IP layer.  In IPv4
 ({{!IPv4=RFC0791}}), the DF bit MUST be set if possible, to prevent
 fragmentation on the path.
 
-Datagrams are required to be of a minimum size under some conditions.  However,
-the size of the datagram is not authenticated.  Therefore, an endpoint MUST NOT
-close a connection when it receives a datagram that does not meet size
-constraints, though the endpoint MAY discard such datagrams.
+QUIC sometimes requires datagrams to be larger than a minimum size; see
+{{version-negotiation}} as an example. However, the size of a datagram is not
```
Comment (kazuho) [inline draft-ietf-quic-transport.md:4131]: I think this sentence is incorrect. The intent of the paragraph is to state that receivers MUST NOT close a connection when it receives a datagram of X bytes carrying a QUIC packet, and if that QUIC packet being carried is required to be sent in a datagram of more htan X bytes.

So maybe something like: _QUIC sometimes requires datagrams to be no smaller than certain size; see {{validate-handshake}} as an example._

Comment (martinthomson) [reply draft-ietf-quic-transport.md:4131]: Or "at least"

**Review by martinthomson — APPROVED** [review]
Other than kazuho's nit  which I regard as minor, this is good.

**Comment by kazuho** [comment]
@janaiyengar I think you missed the change proposed to the reference. It should point to somewhere that imposes minimum datagram size (e.g., {{validate-handshake}}, not {{version-negotiation}}).

**Comment by janaiyengar** [comment]
Urgh. Thanks for catching that, @kazuho. #4343 sent.

**Pull request merged**